shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java

```diff
@@ -107,9 +107,19 @@
      */
     public Collection<SimpleTableSegment> getSimpleTableSegments() {
         Collection<SimpleTableSegment> result = new LinkedList<>();
+        Collection<SimpleTableSegment> tmp = new LinkedList<>();
         for (TableReferenceSegment each: tableReferences) {
             result.addAll(each.getSimpleTableSegments());
         }
+        for (SimpleTableSegment each: result) {
```

Comment:
Maybe you need `a set` to remove duplicated ones? Or do you have anyother solution?

Reply:
Not to remove duplicate elements, but to remove elements that just is alias. @tristaZero

Reply:
Got, if you want to keep those codes, please
Rename `tmp`
Add blank space and rename `each1` in `for (SimpleTableSegment each1:result) `
Or Copy [the deleted codes](https://github.com/apache/incubator-shardingsphere/pull/5154/files) here since they have the same logic.
```
private Collection<SimpleTableSegment> getTableSegments(final Collection<SimpleTableSegment> tableSegments, final JoinedTableContext joinedTable) {
          Collection<SimpleTableSegment> result = new LinkedList<>();
          for (SimpleTableSegment tableSegment : ((CollectionValue<SimpleTableSegment>) visit(joinedTable)).getValue()) {
              if (isTable(tableSegment, tableSegments)) {
                  result.add(tableSegment);
              }
          }
          return result;
      }
 
      private boolean isTable(final SimpleTableSegment owner, final Collection<SimpleTableSegment> tableSegments) {
          for (SimpleTableSegment each : tableSegments) {
              if (owner.getTableName().getIdentifier().getValue().equals(each.getAlias().orElse(null))) {
                  return false;
             }
         }
         return true;
     }
```